onboard/nav/avoidance.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef AVOIDANCE_HPP
+#define AVOIDANCE_HPP
+
+#include "rover.hpp"
+
+class StateMachine;
+
+class Avoidance {
+public:
+    /*************************************************************************/
+    /* Public Member Functions */
+    /*************************************************************************/
+    Avoidance(StateMachine* stateMachine_)
+    : stateMachine(stateMachine_),
+    mJustDetectedObstacle(false) {}
+
+    virtual ~Avoidance() {}
+
+    void updateObstacleAngle( double bearing );
+
+    NavState run( Rover * phoebe, const rapidjson::Document& mRoverConfig );
+
+    virtual Odometry createAvoidancePoint( Rover * phoebe, const double distance ) = 0;
+
+    virtual NavState executeTurnAroundObs( Rover* phoebe, const rapidjson::Document& mRoverConfig ) = 0;
+
+    virtual NavState executeDriveAroundObs( Rover* phoebe ) = 0;
+
```

Comment:
minor: Lots of unneeded whitespace

Reply:
I thought this was the style we were using. Should there be no space between functions in a .hpp file class declaration? Isn't this how it is in every file? The spaces I have are the same as StateMachine.hpp

Reply:
That's not quite what I meant. However, you're right that the other files do it the same way as each other so you can use those as a template.